src/easy.c

```diff
@@ -1482,6 +1482,7 @@ do_curl_setopt(CurlObject *self, PyObject *args)
                 if (PyText_AsStringAndSize(PyTuple_GET_ITEM(listitem, 0), &nstr, &nlen, &nencoded_obj) != 0) {
                     curl_formfree(post);
                     Py_XDECREF(ref_params);
+                    PyText_EncodedDecref(nencoded_obj);
```

Comment:
nencoded_obj will be set to NULL if PyText_AsStringAndSize fails but this doesn't hurt since the PyText_EncodedDecref macro checks it.